### PR TITLE
Fix check lock TTL when getting transaction status

### PR DESCRIPTION
**Pull request by leiysky**

### CODE REVIEW

**Comment by leiysky** [comment]
@solotzg PTAL

File: include/pingcap/kv/LockResolver.h [hunk 1]
```diff
@@ -23,7 +23,24 @@ struct TxnStatus
...
+
+    bool isCacheable()
+    {
+        if (isCommitted())
+        {
+            return true;
+        }
+        if (ttl == 0)
+        {
+            if (action == kvrpcpb::Action::NoAction || action == kvrpcpb::Action::LockNotExistRollback
```
Comment (solotzg) [inline include/pingcap/kv/LockResolver.h:36]: try switch case ?

**Review by solotzg — APPROVED** [review]
rest lgtm

File: include/pingcap/kv/LockResolver.h (hunk 1 above)
Comment (leiysky) [reply include/pingcap/kv/LockResolver.h:36]: This can be expressed by one IF so I think we don't need to use SWITCH.

Comment (hanfei1991) [reply include/pingcap/kv/LockResolver.h:36]: actually switch case is more efficient on runtime.

Comment (solotzg) [reply include/pingcap/kv/LockResolver.h:36]: Maybe for such simple case, compiler itself will make optimization.

**Review by hanfei1991 — APPROVED** [review]
LGTM

**Comment by solotzg** [comment]
need to solve https://github.com/pingcap/tics/issues/1454

**Pull request merged**